### ISSUE TITLE
FPS drop with trackballControls on the new chrome on mobile.

### ISSUE THREAD

**Issue by boulabiar** [mrdoob/three.js#14013]
##### Description of the problem

With the recent Chrome updates on mobile, the FPS drops to 2 FPS when zooming with two fingers.
The FPS for this example is around 60. it drops to 2FPS and creates saccades then get back to normal after the operation. This doesn't happen with single finger operation, only with 2 fingers zoom.
https://threejs.org/examples/misc_controls_trackball

##### Three.js version

- [x] r92

##### Browser

- [X] Chrome mobile

##### OS

- [x] Android

**Comment by Mugen87** [comment]
I've tested the example on my Pixel. Although the performance monitor shows sometimes low frame rates, the experience does not feel jerky but smooth to me. 

BTW: The example does not have an active render loop.

**Comment by boulabiar** [comment]
This is what it looks like on the same version of Chrome 
https://www.youtube.com/watch?v=07VdKlSBIa4

**Comment by boulabiar** [comment]
I have made tests on the OnePlus 3T, and Google (HTC) Nexus 9.
This only happens on the last Chrome mobile versions (at least since 2 months).
Firefox don't have these problems.

**Comment by boulabiar** [comment]
I've made a fix that removed that behavior.
#14017

**Comment by mrdoob** [comment]
> This only happens on the last Chrome mobile versions (at least since 2 months).
Firefox don't have these problems.

Wouldn't this be a Chrome bug?

**Comment by Mugen87** [comment]
@boulabiar Do you mind reporting this here instead: https://bugs.chromium.org/p/chromium/issues/list

**Comment by boulabiar** [comment]
Can you please confirm the issue and star it on chromium?
I tested on another smartphone: Asus Zenphone 2 and it has the fps drop.

@mrdoob you have your Google pixel 9 with you? Can you please try it and confirm the issue on chromium bug reporting site?

**Comment by boulabiar** [comment]
Update: Chrome developers stated that this is a bug, but also an "intentional change".
https://bugs.chromium.org/p/chromium/issues/detail?id=841282

The bug is present and can be seen on every chrome mobile.
So in this case, what to do? Ask Chrome people to undo the policy, or update Trackball controls as they've said by adding |touch-action: none| ?

**Comment by Mugen87** [comment]
> Chrome developers stated that this is a bug, but also an "intentional change".

I don't get this :thinking: 

>  or update Trackball controls as they've said by adding |touch-action: none| ?

Can you commit that adding this piece of CSS solves the issue?

**Comment by boulabiar** [comment]
Yes this line resolved the problem for chrome.
`this.domElement.style.touchAction = 'none';`

is this the file to edit to add it ?
https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js

**Comment by Mugen87** [comment]
This would be the right place:

https://github.com/mrdoob/three.js/blob/a17dc8c8125aab75e25348d0aec5ab4551559b93/examples/js/controls/TrackballControls.js#L14-L15

**Comment by XanderLuciano** [comment]
The info `div` was too large because it had 5px of padding and didn't use `box-sizing: border-box;` so the page renders 10px too wide, and because it's `overflow: hidden` you never notice it.

Properly sizing the HTML elements to the window fixes this issues in all the testing I've done on android. #14622 has the modifications.

**Comment by boulabiar** [comment]
yes, I confirm. The #15042 doesn't fix anything as it shows the lag in the same video pretending to fix it (1:47).
Which commit is merged is not important as long as it fixes the issue and it has been around 6 months since I reported it.

So if anyone can take the change I proposed and commit it, it would be OK.

**Comment by Mugen87** [comment]
> Which commit is merged is not important as long as it fixes the issue and it has been around 6 months since I reported it.

I think @mrdoob is interested in a solution, that works for all controls. If `TrackballControls` is changed, it would be necessary to adjust other controls too in order to retain consistency. Alternatively, we could add `touchAction = none;` to the CSS of all examples that use `TrackballControls`.

**Comment by XanderLuciano** [comment]
`touch-action` is not supported by safari, so this would not completely resolve this issue.

![image](https://user-images.githubusercontent.com/13877593/47351383-b36edf80-d66c-11e8-9d6a-b18bf46fbf51.png)

[MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/touch-action#Browser_compatibility), [caniuse](https://caniuse.com/#feat=touch)

There is a bug tracker for support in safari / webkit: https://bugs.webkit.org/show_bug.cgi?id=133112

There may be other methods to prevent the zoom even though, such as:

```js
window.addEventListener(
    "touchmove",
    function(event) {
        if (event.scale !== 1) {
            event.preventDefault();
        }
    },
    { passive: false }
);
```

from [stack overflow](https://stackoverflow.com/questions/37808180/disable-viewport-zooming-ios-10-safari).

Ideally I'd like to avoid browser specific fixes if possible.

**Comment by Mugen87** [comment]
It seems Safari will support `touch-action` in the upcoming months 🎉

see https://bugs.webkit.org/show_bug.cgi?id=133112#c31

**Comment by Mugen87** [comment]
`touch-action` is now supported in all browsers.

![image](https://user-images.githubusercontent.com/12612165/65813750-8c545280-e1d9-11e9-854f-3e61be753ae0.png)

Besides, `TrackballControls` is now always created like so in the examples and thus not attached to `document` anymore (see #16326):
```js
var controls = new TrackballControls( camera, renderer.domElement );
```
To completely solve the issue, I vote to consider #16336 again. Making `domElement` a mandatory parameter and disallowing `document` as an argument should completely solve this issue.

**Comment by Mugen87** [comment]
Solved via #17612
